### DISCUSSION TITLE
Is spawning hundreds of actors a good idea?

### DISCUSSION THREAD

**Discussion by jaykay-p** [statelyai/xstate#3486] [Q&A] [answered]
hi, first off thanks for this amazing library! 
our team is new to xstate and trying out for implementing a chat service.

Since a message can have different states (pending, sent, ...) and receive many different events, I believe making a message an actor is a right approach. So I made a prototype and tested out. 

When there are about more than 50 messages in a chatroom, the inspector starts to be laggy and difficult to work with during development (application was fine). This led to two questions.

 1. Is spawning hundreds or even thousands of actors okay?
 2. Would there be performance issues when there are hundreds actors communicating?

**Comment by Andarist** [comment, marked as answer]
> When there are about more than 50 messages in a chatroom, the inspector starts to be laggy and difficult to work with during development (application was fine)

There are things that we could do to improve the performance of the inspector. There might be a lot of low-hanging fruits in this area.

> Is spawning hundreds or even thousands of actors okay?

I think that it's fine, a bigger constraint here is memory rather than performance. I'm not saying here that XState comes with a particular big memory footprint - just that the sole notion of spawning actors shouldn't ever be a **performance** issue.

> Would there be performance issues when there are hundreds actors communicating?

It depends on how much traffic are we talking about here but I also don't think this could become a bottleneck in most applications.

**Reply by jaykay-p** [reply]
> There are things that we could do to improve the performance of the inspector.

This would be really great. Thanks!
